plugins/output/flight/flight.go

```diff
@@ -131,11 +99,124 @@ func (f *Flight) Connect() error {
+		case arrow.STRING:
+			metricTag := getTag(metric, schemaField, i)
+			builder.Field(i).(*array.StringBuilder).AppendValues([]string{metricTag}, nil)
+		case arrow.INT32:
+			metricField := getField(metric, schemaField, i)
+			switch value := metricField.(type) {
+			case int32:
+				builder.Field(i).(*array.Int32Builder).AppendValues([]int32{value}, nil)
+			case int64:
+				builder.Field(i).(*array.Int32Builder).AppendValues([]int32{int32(value)}, nil)
+			case float32:
+				builder.Field(i).(*array.Int32Builder).AppendValues([]int32{int32(value)}, nil)
+			case float64:
+				builder.Field(i).(*array.Int32Builder).AppendValues([]int32{int32(value)}, nil)
+			}
+		case arrow.INT64:
+			metricField := getField(metric, schemaField, i)
+			switch value := metricField.(type) {
+			case int32:
+				builder.Field(i).(*array.Int64Builder).AppendValues([]int64{int64(value)}, nil)
+			case int64:
+				builder.Field(i).(*array.Int64Builder).AppendValues([]int64{value}, nil)
+			case float32:
+				builder.Field(i).(*array.Int64Builder).AppendValues([]int64{int64(value)}, nil)
+			case float64:
+				builder.Field(i).(*array.Int64Builder).AppendValues([]int64{int64(value)}, nil)
+			}
+		case arrow.FLOAT32:
+			metricField := getField(metric, schemaField, i)
+			switch value := metricField.(type) {
+			case int32:
+				builder.Field(i).(*array.Float32Builder).AppendValues([]float32{float32(value)}, nil)
+			case int64:
+				builder.Field(i).(*array.Float32Builder).AppendValues([]float32{float32(value)}, nil)
+			case float32:
+				builder.Field(i).(*array.Float32Builder).AppendValues([]float32{value}, nil)
+			case float64:
+				builder.Field(i).(*array.Float32Builder).AppendValues([]float32{float32(value)}, nil)
+			}
+		case arrow.FLOAT64:
+			metricField := getField(metric, schemaField, i)
+			switch value := metricField.(type) {
+			case int32:
+				builder.Field(i).(*array.Float64Builder).AppendValues([]float64{float64(value)}, nil)
+			case int64:
+				builder.Field(i).(*array.Float64Builder).AppendValues([]float64{float64(value)}, nil)
+			case float32:
+				builder.Field(i).(*array.Float64Builder).AppendValues([]float64{float64(value)}, nil)
+			case float64:
+				builder.Field(i).(*array.Float64Builder).AppendValues([]float64{value}, nil)
+			}
+		}
+	}
+}
+
+// Return the value of the metric tag with the name equal to the name of the schema field.
+func getTag(metric telegraf.Metric, schemaField arrow.Field, i int) string {
+	metricTag, wasSet := metric.GetTag(schemaField.Name)
+	if !wasSet {
+		panic(fmt.Sprintf("tag %d : %s not set", i, schemaField.Name))
+	}
+	return metricTag
+}
+
+// Return the value of the metric field with the name equal to the name of the schema field.
+func getField(metric telegraf.Metric, schemaField arrow.Field, i int) interface{} {
+	metricField, wasSet := metric.GetField(schemaField.Name)
+	if !wasSet {
+		panic(fmt.Sprintf("field %d : %s not set", i, schemaField.Name))
```

Comment:
Maybe add a comment to document why `panic()` is used instead of returning an error in `Write()`. I assume it is because there is no reason for Telegraf to try and retransmit the data if a required value is missing.

Reply:
Fixed.

Reply:
Fixed.